posthog/api/notebook.py

```diff
@@ -281,6 +281,8 @@ def _filter_request(self, request: request.Request, queryset: QuerySet) -> Query
                             # replay timestamps are not at the top level, they're one-level down in a content array
                             presence_match_structure = [{"content": [{"type": f"ph-{target}"}]}]
                             id_match_structure = [{"content": [{"attrs": {"sessionRecordingId": match}}]}]
+                        elif target == "query":
```

Comment:
🤔 So we are making `query` only work for a specific Insight? This feels very off...

If this is the way to go then I think the "target" should explicitly be `saved-insight` or something.

Reply:
My understanding was that this is just for cases in which there is an id present e.g. when the search query looks like `query:vk7xz3` where the content after the colon is the short id of the saved insight. You could still search notebooks that contain / do not contain queries with the search param `query:true` or `query:false`.

The only reason we special case this lookup is because the `NotebookNodeQuery` does not have an `id` attribute which the search endpoint assumes. Its id is nested inside the `query` attribute.

Does that explain your question?

Reply:
@pauldambra am I right in thinking that this will still allow you to search for the presence of a query with the `query:true` param?

Reply:
so long as this test is still valid

```
        # filter by insight
        insight_filter_response = self.client.get(f"/api/projects/{self.team.id}/notebooks?contains=query:true")
        assert [n["id"] for n in insight_filter_response.json()["results"]] == [insight_content_notebook]
```

then yep we can filter by presence of "type" and presence of an instance of a "type"